### PR TITLE
added lombok and domain model

### PR DESCRIPTION
**Pull request by alexeyboo**
don't know why these other commits keep chasing me

### CODE REVIEW

File: src/main/java/com/epam/bill/domain/Bill.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.epam.bill.domain;
+
+import com.epam.common.domain.BaseDomain;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data @Builder(toBuilder=true) @NoArgsConstructor
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/bill/domain/Bill.java]: Take a look at the code style here, annotations should be one by one (second below that first)

File: src/main/java/com/epam/booking/domain/Booking.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.epam.common.domain.BaseDomain;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Calendar;
+
+@Data @Builder(toBuilder=true) @NoArgsConstructor
+public class Booking extends BaseDomain {
+    private Calendar start;
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/booking/domain/Booking.java]: Why we use Calendar here? In java 8 we have good LocalDate classes

File: src/main/java/com/epam/hotel/domain/Hotel.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data @Builder(toBuilder=true) @NoArgsConstructor
+public class Hotel extends BaseDomain {
+    private String name;
+    private String location;
+    private boolean luxury;
```
Comment (imbirevkotlya) [inline src/main/java/com/epam/hotel/domain/Hotel.java]: I believe, luxury should be int?

File: src/main/java/com/epam/hotel/domain/Hotel.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    private List<Room> allHotelRooms;
```
[the remark this answers is not in the document]
Comment (alexeyboo) [reply src/main/java/com/epam/hotel/domain/Hotel.java]: and what about services or repoes then? all in one package?

File: src/main/java/com/epam/common/domain/BaseDomain.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.epam.common.domain;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data @Builder(toBuilder=true) @NoArgsConstructor
+public abstract class BaseDomain {
```
Comment (IlyasYOY) [inline src/main/java/com/epam/common/domain/BaseDomain.java]: Do you need this class?

File: pom.xml [hunk 6]
```diff
@@ -15,5 +15,12 @@
             <version>4.12</version>
             <scope>test</scope>
         </dependency>
+      <<<<<<< master
```
Comment (IlyasYOY) [inline pom.xml]: Remove this.

File: src/main/java/com/epam/hotel/domain/Hotel.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (IlyasYOY) [reply src/main/java/com/epam/hotel/domain/Hotel.java]: allHotelRooms?
maybe simply rooms?

File: src/main/java/com/epam/user/domain/User.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data @Builder(toBuilder=true) @NoArgsConstructor
+public class User extends BaseDomain {
+    private Role role;
+    private String login;
+    private String password;
+    private List<Bill> userBills;
```
Comment (IlyasYOY) [inline src/main/java/com/epam/user/domain/User.java]: bills

it obvious from the context that they belong to user

File: pom.xml [hunk 8]
```diff
@@ -6,6 +6,18 @@
     <groupId>com.epam</groupId>
     <artifactId>2019-spring-epam-first-project</artifactId>
     <version>1.0-SNAPSHOT</version>
+    <build>
```
Comment (imbirevkotlya) [inline pom.xml]: please, delete this build tag info from the request, our Travis wants to work with java 8

**Pull request merged**